Real-time engine tweaking via :angel:Script - with example.

ohlidalp opened the pull request:
Added input events `EV_TRUCK_MANUAL_CLUTCH_MODIFIER_25/50` To go with existing accel/brake modifiers. Suggested by Gibbzy on Discord: https://discord.com/channels/136544456244461568/189904947649708032/1280998672246771753

UPDATE: Turns out this needs more research - the modifiers work but they don't have the desired effect, see Discussion: https://discord.com/channels/136544456244461568/189904947649708032/1282776986107773010. To research further, I've dusted off my Engine-AngelScript bindings (made a year ago and never merged) and created an engine diagnostic script 📜 'example_actor_engineDiag.as'.

UPDATE2: To help modders figure out what the engine needs, I've extended the above script with real-time engine tweaking capabilities. This is possible thanks to the `ACTORSIMATTR_` system introduced recently in https://github.com/RigsOfRods/rigs-of-rods/pull/3210. 'engturbo' isn't in the tool yet but the ACTORSIMATTR_ constants exist.
Either way it's just an example script, if anybody finds it almost useful let me know.
![image](https://github.com/user-attachments/assets/0d17a5e5-d3ea-41c2-9a9f-468247154d1a)

ohlidalp commented:
Didn't work as intended, commentary and research: https://discord.com/channels/136544456244461568/189904947649708032/1282776986107773010

I dusted off some AngelScript-engine binding code (1 year old) and created a simple engine diag script. Currently it only shows engine/engoption settings and current engine state as seen by the game. To run it, open in-game console (hotkey ~ or topmenu/Tools/Console) and say `loadscript example_game_engineDiag.as`

CuriousMike56 reviewed (commented):
Awesome script. One issue though: 

Spawn the [Pontiac Firebird](https://forum.rigsofrods.org/resources/1970-73-pontiac-firebird-restomod-protouring.1153/) and switch to neutral gear, results in a divide by zero error:

```
The script ended with exception; details below:
	context.ExceptionLineNumber: 38
	context.ExceptionString: Divide by zero
	context.ExceptionFunction.Declaration: void updateEnginePlotBuffers(EngineClass@)
	context.ExceptionFunction.ModuleName: example_actor_engineDiag.as(category:CUSTOM,unique ID:11)
	context.ExceptionFunction.ScriptSectionName: example_actor_engineDiag.as
	context.ExceptionFunction.ObjectName: 
```

ohlidalp commented:
I've renamed confusing EngineClass attr getters:
* `getMaxRPM()` --> `getShiftDownRPM()` -- equivalent of `ACTORSIMATTR_ENGINE_SHIFT_DOWN_RPM`
* `getMinRPM()` --> `getShiftUpRPM()` -- equivalent of `ACTORSIMATTR_ENGINE_SHIFT_UP_RPM`

The 'example_actor_engineDiag.as' was updated:
* updated 'engine' tab to show new names
* completely removed the 'clutch' tab as it didn't make sense anymore - I got confused by the min/max RPM when writing it. @CuriousMike56 This also fixes the division-by-zero script crash with 1970 Pontiac Firebird.
* fixed plot min/max values in the 'state' tab.
